### ISSUE TITLE
@defer isn't working properly with style and http requests

### ISSUE THREAD

**Issue by robertIsaac** [angular/angular#51973]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

when using the new `@defer` I notice that two examples aren't working and I'm not sure why
1. when loading data using http, while the http request does appear in the browser network it's content never show up in the dom
2. the defer component styles aren't implemented at all

### Please provide a link to a minimal reproduction of the bug

https://github.com/robertIsaac/first-look-at-angular-defer

### Please provide the exception or error you saw

```true
you will find in the repo two branches called `before-defer` and `master`
I expect the two example have the same behavior after and before defer
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 17.0.0-next.6
Node: 18.16.0
Package Manager: npm 9.6.7
OS: win32 x64

Angular: 17.0.0-next.6
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router, ssr

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.0-next.6
@angular-devkit/build-angular   17.0.0-next.6
@angular-devkit/core            17.0.0-next.6
@angular-devkit/schematics      17.0.0-next.6
@angular/cdk                    17.0.0-next.5
@angular/material               17.0.0-next.5
@schematics/angular             17.0.0-next.6
rxjs                            7.8.1
typescript                      5.2.2
zone.js                         0.14.0
```

### Anything else?

I understand this is still a work in progress, I just wanted to give you test cases to help with the development of this great work you are doing

**Comment by AndrewKushnir** [comment]
@robertIsaac thanks for reporting the issue and providing a repro 👍 

I've performed the investigation and found out that the problem was caused by a missing change-detection round that was not scheduled after triggering a block. The issue is that the `requestIdleCallback` is not patched by Zone.js and as a result, the callback is not invoked within a zone. I've created a PR https://github.com/angular/angular/pull/51975 to resolve the problem.

The mentioned PR should resolve the issue with HTTP calls, but I'm not sure I fully understand the second issue ("the defer component styles aren't implemented at all"), so I was unable to verify. Could you please provide some additional information about this issue (expected vs actual behavior)?

Thank you.

**Comment by robertIsaac** [comment]
@AndrewKushnir thanks for clearing the http part
for the second part here is what I expect
![image](https://github.com/angular/angular/assets/14242054/9e33e9b6-7116-4e3c-81f8-68defe5b22bb)

here is actually how it looks like with defer
![image](https://github.com/angular/angular/assets/14242054/d85d2caf-8b79-493e-ad8e-c482f8ab260d)

to see why I expect it like that just remove the `@defer` from `src/app/bundle-size-example/bundle-size-example.component.html`
it seems `color="primary"` isn't take an affect also `src/app/add-comment/add-comment.component.scss` isn't loaded at all

**Comment by JeanMeche** [comment]
@robertIsaac It's the same ChangeDetection issue. If you add a `setInterval(() => {}, 100)` (which fires CD every 100ms) on your `BundleSizeExampleComponent` you will get the expected result. 

Andrews's fix on #51975 should fix both issues !

**Comment by robertIsaac** [comment]
> @robertIsaac It's the same ChangeDetection issue. If you add a `setInterval(() => {}, 100)` (which fires CD every 100ms) on your `BundleSizeExampleComponent` you will get the expected result.
> 
> Andrews's fix on #51975 should fix both issues !

okay I see, I tried to trigger it in ngOnInit, but didn't work so I thought it's different

**Comment by AndrewKushnir** [comment]
@robertIsaac FYI the fix for this issue was merged today and it will be included into the upcoming 17.0.0-next.7 release this week. Thank you.
